panic when editing an existing rule

**Describe the bug** When editing an existing policy rule, a panic can be induced

**To Reproduce** Steps to reproduce the behavior:

1. Start pomerium with a helm chart and a single policy rule
2. Edit configmap to change protocol of destination from http:// to https://
3. Panic:

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa438d4]

goroutine 38 [running]:
github.com/pomerium/pomerium/authorize.(*Authorize).UpdateOptions(0x0, 0x0, 0xc4cbfe, 0x5, 0xc00002c04e, 0x2c, 0xc00002e009, 0x5, 0xc4d369, 0x6, ...)
	/go/src/github.com/pomerium/pomerium/authorize/authorize.go:61 +0xb4
main.handleConfigUpdate(0xc00028e000, 0xc000377e78, 0x2, 0x2, 0x18)
	/go/src/github.com/pomerium/pomerium/cmd/pomerium/main.go:233 +0x2a1
main.main.func1(0xc000264ee0, 0x18, 0x8)
	/go/src/github.com/pomerium/pomerium/cmd/pomerium/main.go:70 +0x10b
github.com/spf13/viper.(*Viper).WatchConfig.func1.1(0xc0000856d0, 0xc00002f370, 0x7ffd17864bff, 0x19, 0x7ffd17864bff, 0x19, 0xc0001971a0, 0xc0000ce300)
	/go/pkg/mod/github.com/spf13/viper@v1.4.0/viper.go:328 +0x232
created by github.com/spf13/viper.(*Viper).WatchConfig.func1
	/go/pkg/mod/github.com/spf13/viper@v1.4.0/viper.go:306 +0x2a1
```

**Expected behavior**

Clean policy reload

**Environment:**

- Pomerium 0.1.0
- Helm chart 1.2.2
- Kubernetes / Docker-for-Desktop / OSX / AMD64

Configuration file(s):

```text
  policy:
  - from: https://grafana.lan.XXXXX
    to: http://httpbin.org
    allowed_domains:
      - XXXXX
```

Logs(s):

```text
{"level":"info","file":"/etc/pomerium/..data_tmp","time":"2019-08-01T13:31:53Z","message":"cmd/pomerium: configuration file changed"}
{"level":"debug","old-checksum":"6feab29bf832b12b","new-checksum":"b62f2ca3151c7612","time":"2019-08-01T13:31:53Z","message":"cmd/pomerium: configuration file changed"}
{"level":"info","checksum":"b62f2ca3151c7612","time":"2019-08-01T13:31:53Z","message":"cmd/pomerium: checksum changed"}
{"level":"info","time":"2019-08-01T13:31:53Z","message":"authorize: updating options"}
{"level":"debug","route":"grafana.lan.XXXXX","domain":"XXXXX","time":"2019-08-01T13:31:53Z","message":"add domain"}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,7 @@
 ### FIXED
 
 - Fixed potential race condition when signing requests. [GH-240]
+- Fixed panic when reloading configuration in single service mode [GH-247]
 
 ## v0.1.0
 

--- a/authorize/authorize.go
+++ b/authorize/authorize.go
@@ -61,6 +61,9 @@ func (a *Authorize) ValidIdentity(route string, identity *Identity) bool {
 // UpdateOptions updates internal structures based on config.Options
 func (a *Authorize) UpdateOptions(o config.Options) error {
 	log.Info().Msg("authorize: updating options")
+	if a == nil {
+		return nil
+	}
 	a.identityAccess = NewIdentityWhitelist(o.Policies, o.Administrators)
 	return nil
 }

--- a/authorize/authorize_test.go
+++ b/authorize/authorize_test.go
@@ -94,4 +94,8 @@ func Test_UpdateOptions(t *testing.T) {
 			}
 		})
 	}
+
+	// Test nil
+	var a *Authorize
+	a.UpdateOptions(config.Options{})
 }

--- a/proxy/proxy.go
+++ b/proxy/proxy.go
@@ -321,5 +321,8 @@ func (p *Proxy) newReverseProxyHandler(rp *httputil.ReverseProxy, route *config.
 
 // UpdateOptions updates internal structures based on config.Options
 func (p *Proxy) UpdateOptions(o config.Options) error {
+	if p == nil {
+		return nil
+	}
 	return p.UpdatePolicies(&o)
 }

--- a/proxy/proxy_test.go
+++ b/proxy/proxy_test.go
@@ -329,4 +329,8 @@ func Test_UpdateOptions(t *testing.T) {
 			}
 		})
 	}
+
+	// Test nil
+	var p *Proxy
+	p.UpdateOptions(config.Options{})
 }

